spec/classes/root_user_spec.rb

```diff
@@ -27,6 +27,16 @@
         }
       end
 
+      context 'with password' do
+        let(:params) {{
+          :password => 'mysecretpassword'
+        }}
+        it {
+          is_expected.to create_file('/root')
+          is_expected.to create_user('root').with_password('mysecretpassword')
```

Comment:
This should technically fail since it would create an invalid password entry on the target host.

Reply:
Fixed the way the tests were laid out and ensure we're getting undef or a hash in various scenarios.